packages/charts/src/components/TimelineChart/TimelineChartAnnotation.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { CSSProperties } from 'react';
+import { DEFAULT_ROW_HEIGHT } from './util/constants';
+
+interface TimelineChartAnnotationProps {
+  /**
+   * The width of the annotation. If not provided, it expands to fill
+   * the timeline.
+   */
+  width?: number;
+
+  /**
+   * The height of the annotation. Preferably, this would be the same height as the
+   * rowHeight of the chart. A different height could be set if the annotation
+   * spans multiple rows. If not set, the __rowHeight__  specified for the
+   * `TimelineChartAnnotation` is used.
+   */
+  height?: string;
+
+  /**
+   * The index of the row the annotation should be applied to. The top of the anotation is
+   * aligned with the top of the row.
+   * @default 0
+   */
+  rowIndex?: number;
+
+  /**
+   * The row height of the chart. This should **always** be set to the same value as
+   * the rowHeight of the chart in order for the top of the annotation to be aligned
+   * with the top of the row the annotation is applied to. If not set, the default
+   * __rowHeight__ of the `TimelineChart` is used.
+   * @default 25
+   */
+  rowHeight?: number;
+
+  /**
+   * The diagram, symbol or illustration that should be in the annotation. If one
+   * is planning to do a lot of custom annotations on a row, it is probably
+   * advised to use the HTML Canvas for drawing due to performance reasons. And
+   * it is also better to draw all the annotations on one row in the same figure.
+   */
+  figure: JSX.Element;
+}
+
+/**
+ * This is designed to be used for creating custom annotations, markers
+ * or illustrations on the timeline of the chart.
+ */
+const TimelineChartAnnotation: React.FC<TimelineChartAnnotationProps> = ({
+  width,
+  height,
+  rowIndex,
+  rowHeight,
+  figure
+}) => {
+  rowHeight = rowHeight ?? DEFAULT_ROW_HEIGHT;
+  rowIndex = rowIndex ?? 0;
+
+  const style: CSSProperties = {
+    width: width != null ? `${width}px` : 'auto',
+    height: height != null ? height : rowHeight,
+    position: 'absolute',
+    top: `${rowIndex * rowHeight}px`
+  };
+
+  return (
+    <div className="timeline-chart-annotation" style={style}>
```

Comment:
Hi @Lukas742,

I have started working on the suggestions in the review 2 days ago. The only one that I still don't have a good solution for are these classes I added to some of the subcomponents. I am using the classes as a querySelector in the tests to be able to check the behavior of the component under certain conditions. I suppose I could just add a data-testId and use a getByTestId in the test too but that is probably the same thing. Do you have an alternative idea? Thanks

Reply:
First of all, sorry that I haven't continued my review. It has been a pretty busy week. I'll try to get to it in the beginning of next week.

We usually don't add attributes or classes just for testing purposes to our components. In this case it should be sufficient to add e.g. a `data-testid` data attribute to the `TimelineChartAnnotation` component that is rendered in the test env, and then you'll be able to query it by using [getByTestId](https://testing-library.com/docs/react-testing-library/cheatsheet) of react-testing-library.

If an element is not exposed at the top level and data attributes cannot be added via React props, then we usually add a `data-component-name` attribute, describing the element. `component-name` is actually the wrong wording for this, but since this is used in many components, we sticked to it to stay consistent. You can search our project for this attribute to see how we usually name elements like this. [Example1](https://github.com/SAP/ui5-webcomponents-react/blob/dfbf91fb384313d224d618079268f6abcd73208a/packages/main/src/components/AnalyticalTable/TableBody/VirtualTableBody.tsx#L97), [Example2](https://github.com/SAP/ui5-webcomponents-react/blob/dfbf91fb384313d224d618079268f6abcd73208a/packages/main/src/components/AnalyticalTable/TableBody/VirtualTableBodyContainer.tsx#L94)